Environment variables documentation

MukuFlash03 opened the pull request:
Creating a consolidated documentation for environment variables that are used across the repositories of OpenPATH repo.

Initially starting out with repositories involved in the Redesign changes ([PR](https://github.com/e-mission/e-mission-server/pull/961/), [Issue](https://github.com/e-mission/e-mission-docs/issues/1048)) - `e-mission-server, join-page, admin-dash, public-dash`

shankari commented on docs/dev/back/config_environment_variables.md:
the whole point of having searches instead of listing out a table was because the table will quickly become obsolete and gives people a false sense of knowledge. Please remove.

shankari reviewed (changes requested):
@MukuFlash03 I wasn't complaining about the format (table versus list) in which the existing variables are listed. I think we should not list the variables at all. The list of variables will become obsolete as the code changes, and at this point, we are not signing up to keep it maintained.

As a concrete example, I just checked in https://github.com/e-mission/e-mission-server/pull/980 so the list of push variables in this PR is already obsolete. This should either be autogenerated, or it should be removed entirely.

MukuFlash03 commented:
Right, I did see the Push config changes and yes it would keep rendering the doc obsolete. I had this in mind when I removed the table. But in the flow of updating the documentation, I just missed the whole point and re-added it as a list. 

Removed references of individual variable names. Search results mentioned only now.

shankari commented on docs/dev/back/config_environment_variables.md:
```diff
@@ -0,0 +1,15 @@
+
+#### Overpass, Nominatim, Geofabrik
+
+- Use this [search](https://github.com/search?q=repo%3Ae-mission%2Fe-mission-server+%2F.environ.get%5C%28%5C%22%5BA-Z%5D%2F&type=code) to obtain usage of `Overpass, Nominatim, Geofabrik` environment variables in the [e-mission-server](https://github.com/e-mission/e-mission-server) repository.
+- Use this [search](https://github.com/search?q=repo%3Ae-mission%2Fe-mission-server+geofabrik&type=code) to view some variables that are set using GitHub secrets in the [e-mission-server](https://github.com/e-mission/e-mission-server) GitHub workflows.
```
what is the difference between these two searches. the second seems to look for `geofabrik`, but you also list `geofabrik` in the first search.

MukuFlash03 replied:
With reference to Geofabrik, the 1st search lists `GEOFABRIK_OVERPASS_KEY`, `GEOFABRIK_QUERY_URL`.

In addition to these, the 2nd search also lists `GFBK_KEY` which is used as a suffix to `GEOFABRIK_QUERY_URL` in [emission/integrationTests/docker-compose.yml](https://github.com/e-mission/e-mission-server/blob/204cec1795435eed9355c29d2c869e08bfddf673/emission/integrationTests/docker-compose.yml#L11).
This variable is set while running the docker compose file in a workflow file using a GitHub secret `GEOFABRIK_API` which is different from `GEOFABRIK_OVERPASS_KEY`.

This doesn't show up while searching for `environ.get()` since we are using it directly in the docker-compose file.

shankari replied:
Hm. I am not sure we need to list the secrets for GitHub workflows, given that we don't really want people to be working on forks. However, I guess it could be useful at some point. The problem, though, is that if we add another secret to the GH workflow, it won't show up with this search.

We have gone back and forth enough times on this that I think that I am not going to force this change. But I fully expect it to bitrot within the next 5 years, at which point we should remove it. Let's see if I am correct!

shankari commented:
Squash-merging to avoid commit churn
